src/DotNetWorker.Core/Context/Features/DefaultInputConversionFeature.cs

```diff
@@ -0,0 +1,116 @@
+        public async ValueTask<ConversionResult> ConvertAsync(ConverterContext converterContext)
+        {
+            // Check a converter is explicitly specified via the converter context. If so, use that.
+            IInputConverter? converterFromContext = GetConverterFromContext(converterContext);
+
+            if (converterFromContext != null)
+            {
+                var conversionResult = await ConvertAsyncUsingConverter(converterFromContext, converterContext);
+
+                if (conversionResult.IsSuccess)
+                {
+                    return conversionResult;
+                }
+            }
+
+            // Check the default converters and see which one can successfully convert.
+            // The first converter to successfully convert wins.
+            // For example, this allows a converter that parses JSON strings to return false if the
+            // string is not valid JSON. This manager will then continue with the next matching provider.
+            foreach (var converter in _inputConverterProvider.DefaultConverters)
+            {
+                var conversionResult = await ConvertAsyncUsingConverter(converter, converterContext);
+
+                if (conversionResult.IsSuccess)
+                {
+                    return conversionResult;
+                }
+            }
+
+            return default;
+        }
+                
+        private ValueTask<ConversionResult> ConvertAsyncUsingConverter(IInputConverter converter, ConverterContext context)
+        {
+            var conversionResultTask = converter.ConvertAsync(context);
+
+            if (conversionResultTask.IsCompletedSuccessfully)
+            {
+                return new ValueTask<ConversionResult>(conversionResultTask.Result);
+            }
+
+            return AwaitAndReturnConversionTaskResult(conversionResultTask);
+        }
+
+        private async ValueTask<ConversionResult> AwaitAndReturnConversionTaskResult(ValueTask<ConversionResult> conversionResultTask)
+        {
+            var result = await conversionResultTask;
+
+            return result;
+        }
+
+        /// <summary>
+        /// Gets an <see cref="IInputConverter"/> instance if converter context has information about what converter to be used.
+        /// </summary>
+        /// <param name="context">The converter context.</param>
+        /// <returns>An IInputConverter instance or null</returns>
+        private IInputConverter? GetConverterFromContext(ConverterContext context)
+        {
+            Type? converterType = default;
+
+            // Check a converter is specified on the conversionContext.Properties. If yes,use that.
+            if (context.Properties != null
+                && context.Properties.TryGetValue(PropertyBagKeys.ConverterType, out var converterTypeFullAssemblyNameObj))
+            {
+                converterType = Type.GetType(converterTypeFullAssemblyNameObj.ToString());
```

Comment:
A potential optimization is a cache for this fully qualified assembly name to the type so that we do not need to make `GetType` every time.